app/workers/share_analytics_updater.rb

```diff
@@ -0,0 +1,46 @@
+require "net/http"
+require "uri"
+
+class ShareAnalyticsUpdater
+  def self.update_shares
+    new(nil).update_collection
+  end
+
+  def initialize(collection)
+    #Collection can be passed explicitly in order to reuse this task to update only a section of buttons at a later time.
+    @collection = collection || Share::Button.all
+    @uri = URI.parse('http://run.shareprogress.org/api/v1/buttons/analytics')
+    @sp_api_key = ENV['SHARE_PROGRESS_API_KEY']
+  end
+
+  def update_collection
+    update_shares
+  end
+
+  private
+
+  def update_shares
+    @collection.each do |button|
+      begin
+        request_analytics_from_sp(button)
+      rescue ::ShareProgressApiError
+        puts $!.message
```

Comment:
When an exception is raised, Ruby places a reference to the Exception object associated with the exception into the global variable`$!`. In this case, this is a `ShareProgressApiError` that has a message associated with it (basically I just print whatever the ShareProgress api sent back in `response.message`).